Documentation/proposals/202501-configuration-object-status-subresource.md

```diff
@@ -0,0 +1,291 @@
+
+##### Alternative (Not recommended)
+
+An alternative approach would involve allowing the workload's ServiceAccount to update its own binding in the status subresource of the configuration resource. In this method, the job_name field could be used to infer a reference to the originating resource—such as ServiceMonitor/namespace/name/index.
+
+Cons:
+* The workload's ServiceAccount would need additional permissions to update its own resource’s status subresource.
+* Complexity in Resource Mapping.
+
+#### How to remove a binding when the workload is deleted ?
+
+When a workload resource is created, we add a finalizer to it to ensure proper cleanup before deletion. If a user later requests deletion of the resource, Kubernetes does not immediately remove it; instead, it sets a deletionTimestamp on the resource. This triggers an update event, which the controller receives and processes. The controller checks if the deletionTimestamp is set to determine if the resource is in the process of being deleted. If so, the controller proceeds to clean up associated references from relevant configuration resources (e.g., ServiceMonitors, PrometheusRules). Once the cleanup is complete, the controller removes the finalizer from the workload resource, allowing Kubernetes to complete the deletion process.
+
+#### How to remove invalid bindings from config-resources status ?
+
+A dedicated goroutine runs continuously to monitor configuration resources for any invalid bindings.
+
+## Alternatives
+
+#### Dedicated CRD Approach for Configuration-Workload Mapping
+
+A potential solution to mapping a configuration resource to a workload resource is the introduction of a Custom Resource Definition (CRD). This new CRD would act as an intermediary, maintaining a clear association between configurations and workloads.
+
+The [Secrets Store CSI Driver](https://secrets-store-csi-driver.sigs.k8s.io/introduction) handles the Secrets Provider pod status in a similar way.
+
+It comes with the following drawbacks:
+* Introducing a new CRD could lead to additional operational complexity.
+* It requires installation, maintenance and versioning, which could increase the administrative burden.
+
+#### Storing Information in the Workload Resource
+
+Another approach is to store configuration mappings directly within the workload resource.
+
+It comes with the following drawbacks:
+* Workload resources could reference a high number of configuration resources (it isn't uncommon for a Prometheus resource to select more than a hundred of service monitors + pod monitors + rules).
+* Storing all these mappings within a single workload resource could lead to excessive API payload sizes.
+* Owners of configuration resources not having permissions to view the workload resource won’t have a view of the status which is one of the main goals of this effort.
+
+## Action Plan
+
+...
```

Comment:
A rough implementation plan would be nice to have (it's ok to only list the main steps that you'll plan to tackle, e.g. don't list refactoring tasks).